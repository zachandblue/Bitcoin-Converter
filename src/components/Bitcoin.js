import React, { Component } from "react";

class Bitcoin extends Component {
  render() {
    return (
      <div>
        <div className="exchange-rate-label">
          <h3>Rate</h3>
        </div>
        <h3 className="rate">
          <div className="exchange-rate">
            {(1 * this.props.exchangeRate).toFixed(2)}
          </div>
        </h3>
        <div className="bitcoin-value-label">
          <h3>Bitcoins</h3>
        </div>
        <h1 className="bitcoin-value">
          <div className="bitcoin">
            {(this.props.dollars / this.props.exchangeRate).toFixed(2)}
          </div>
        </h1>
      </div>
    );
  }
}

export default Bitcoin;
