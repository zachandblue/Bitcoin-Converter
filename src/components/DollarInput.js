import React, { Component } from "react";

export class DollarInput extends Component {
  constructor() {
    super();

    this.state = {
      dollars: ""
    };
  }
  render() {
    return (
      <div className="dollar-input">
        <form>
          <div className="dollars">
            <h3>US Dollars</h3>
          </div>
          <input
            type="text"
            placeholder="$"
            className="dollar-input"
            onChange={this.props.handleChange}
            value={this.props.dollars}
          />
        </form>
      </div>
    );
  }
}

export default DollarInput;
