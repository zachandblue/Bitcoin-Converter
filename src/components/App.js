import React, { Component } from "react";
import DollarInput from "./DollarInput";
import Bitcoin from "./Bitcoin";
import getData from "../api";

export class App extends Component {
  constructor() {
    super();

    this.state = {
      dollars: "",
      exchangeRate: "",
      bitcoins: ""
    };

    this.apiCall = this.apiCall.bind(this);
  }

  handleChange = e => {
    const dollars = e.target.value;
    this.setState({ dollars });
  };

  apiCall = async () => {
    const result = await getData();
    const exchangeRate = result.bpi.USD.rate_float;
    this.setState({ exchangeRate });
    return true;
  };

  componentDidMount = () => {
    this.apiCall();
  };
  render() {
    return (
      <div className="main">
        <div className="header">
          <h1>Cash To Bitcoin</h1>
          <i className="fab fa-bitcoin" />
          <div className="header-triangle-1" />
          <div className="header-triangle-2" />
        </div>

        <div className="app">
          <div className=" main-box-1" />
          <DollarInput
            handleChange={this.handleChange}
            dollars={this.state.dollars}
          />

          <Bitcoin
            exchangeRate={this.state.exchangeRate}
            dollars={this.state.dollars}
            exchangeRate={this.state.exchangeRate}
          />
          <div className=" main-box-2" />
          <div className=" main-box-3" />
          <div className=" main-box-4" />
          <div className=" main-box-5" />
          <div className=" main-box-6" />
          <div className="credits">
            Powered by <a href="http://www.coindesk.com/price">CoinDesk</a>
          </div>
        </div>

        <div className="footer">
          {/* <div className="footer-triangle-1" /> */}
        </div>
      </div>
    );
  }
}

export default App;
