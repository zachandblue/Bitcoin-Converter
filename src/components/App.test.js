import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import Bitcoin from "./Bitcoin";
import DollarInput from "./DollarInput";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import getData from "../api";

describe("App", () => {
  const app = shallow(<App />);
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });

  it("has child component `DollarInput`", () => {
    expect(app.find("DollarInput").exists()).toBe(true);
  });
  it("initializes `state` with empty `dollars`, `exchangeRate`, and `bitcoins` keys", () => {
    expect(app.state()).toEqual({
      dollars: "",
      exchangeRate: "",
      bitcoins: ""
    });
  });

  // it("should do what I like", () => {
  //   app.instance().apiCall = jest.fn();
  //   expect(app.instance().apiCall).toMatchSnapshot();
  // });

  it("calls getData function when the component mounts", () => {
    app.instance().apiCall = jest.fn();

    app.update();
    app.instance().componentDidMount();
    expect(app.instance().apiCall).toHaveBeenCalledTimes(1);
  });

  // it("returns the expected result", () => {
  //   app.instance().apiCall = jest.fn();
  //   app.update();
  //   app.instance().componentDidMount();
  //   expect(app.instance().apiCall()).toBe(true);
  // });

  describe("when the `getData` function is called in `ComponentDidMount`", () => {
    let result;
    const mock = new MockAdapter(axios);
    const sample = { data: { bpi: { USD: { rate_float: 5 } } } };

    beforeEach(async () => {
      mock
        .onGet("https://api.coindesk.com/v1/bpi/currentprice.json")
        .reply(200, sample);
      result = await getData();
    });

    it("calls bitcoin API to retrieve exchange rate", async () => {
      expect(result).toEqual(sample);
    });

    it("updates the state of `exchangeRate` to be the result of the API call", async () => {
      const exchangeRate = result.data.bpi.USD.rate_float;
      const mounted = shallow(<Bitcoin exchangeRate={exchangeRate} />);
      expect(parseInt(mounted.find(".exchange-rate").text()), 10).toEqual(
        parseInt(exchangeRate, 10)
      );
    });
  });

  describe("when `handleChange` is called", () => {
    it("updates the `dollars` state", () => {
      const dollars = 100;
      const e = { target: { value: dollars } };

      app.instance().handleChange(e);

      expect(app.state().dollars).toEqual(dollars);
    });
  });
});
