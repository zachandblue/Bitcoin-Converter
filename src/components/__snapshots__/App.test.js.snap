// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <h1>
                    Cash To Bitcoin
          </h1>
          <i
                    className="fab fa-bitcoin"
          />
          <div
                    className="header-triangle-1"
          />
          <div
                    className="header-triangle-2"
          />
</div>,
        <div
          className="app"
>
          <div
                    className=" main-box-1"
          />
          <DollarInput
                    dollars=""
                    handleChange={[Function]}
          />
          <Bitcoin
                    dollars=""
                    exchangeRate=""
          />
          <div
                    className=" main-box-2"
          />
          <div
                    className=" main-box-3"
          />
          <div
                    className=" main-box-4"
          />
          <div
                    className=" main-box-5"
          />
          <div
                    className=" main-box-6"
          />
          <div
                    className="credits"
          >
                    Powered by 
                    <a
                              href="http://www.coindesk.com/price"
                    >
                              CoinDesk
                    </a>
          </div>
</div>,
        <div
          className="footer"
/>,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Cash To Bitcoin
</h1>,
            <i
              className="fab fa-bitcoin"
/>,
            <div
              className="header-triangle-1"
/>,
            <div
              className="header-triangle-2"
/>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cash To Bitcoin",
            },
            "ref": null,
            "rendered": "Cash To Bitcoin",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fab fa-bitcoin",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-triangle-1",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-triangle-2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className=" main-box-1"
/>,
            <DollarInput
              dollars=""
              handleChange={[Function]}
/>,
            <Bitcoin
              dollars=""
              exchangeRate=""
/>,
            <div
              className=" main-box-2"
/>,
            <div
              className=" main-box-3"
/>,
            <div
              className=" main-box-4"
/>,
            <div
              className=" main-box-5"
/>,
            <div
              className=" main-box-6"
/>,
            <div
              className="credits"
>
              Powered by 
              <a
                            href="http://www.coindesk.com/price"
              >
                            CoinDesk
              </a>
</div>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " main-box-1",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dollars": "",
              "handleChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dollars": "",
              "exchangeRate": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " main-box-2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " main-box-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " main-box-4",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " main-box-5",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": " main-box-6",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Powered by ",
                <a
                  href="http://www.coindesk.com/price"
>
                  CoinDesk
</a>,
              ],
              "className": "credits",
            },
            "ref": null,
            "rendered": Array [
              "Powered by ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CoinDesk",
                  "href": "http://www.coindesk.com/price",
                },
                "ref": null,
                "rendered": "CoinDesk",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "footer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <h1>
                        Cash To Bitcoin
            </h1>
            <i
                        className="fab fa-bitcoin"
            />
            <div
                        className="header-triangle-1"
            />
            <div
                        className="header-triangle-2"
            />
</div>,
          <div
            className="app"
>
            <div
                        className=" main-box-1"
            />
            <DollarInput
                        dollars=""
                        handleChange={[Function]}
            />
            <Bitcoin
                        dollars=""
                        exchangeRate=""
            />
            <div
                        className=" main-box-2"
            />
            <div
                        className=" main-box-3"
            />
            <div
                        className=" main-box-4"
            />
            <div
                        className=" main-box-5"
            />
            <div
                        className=" main-box-6"
            />
            <div
                        className="credits"
            >
                        Powered by 
                        <a
                                    href="http://www.coindesk.com/price"
                        >
                                    CoinDesk
                        </a>
            </div>
</div>,
          <div
            className="footer"
/>,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Cash To Bitcoin
</h1>,
              <i
                className="fab fa-bitcoin"
/>,
              <div
                className="header-triangle-1"
/>,
              <div
                className="header-triangle-2"
/>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cash To Bitcoin",
              },
              "ref": null,
              "rendered": "Cash To Bitcoin",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fab fa-bitcoin",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "header-triangle-1",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "header-triangle-2",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className=" main-box-1"
/>,
              <DollarInput
                dollars=""
                handleChange={[Function]}
/>,
              <Bitcoin
                dollars=""
                exchangeRate=""
/>,
              <div
                className=" main-box-2"
/>,
              <div
                className=" main-box-3"
/>,
              <div
                className=" main-box-4"
/>,
              <div
                className=" main-box-5"
/>,
              <div
                className=" main-box-6"
/>,
              <div
                className="credits"
>
                Powered by 
                <a
                                href="http://www.coindesk.com/price"
                >
                                CoinDesk
                </a>
</div>,
            ],
            "className": "app",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": " main-box-1",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "dollars": "",
                "handleChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "dollars": "",
                "exchangeRate": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": " main-box-2",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": " main-box-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": " main-box-4",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": " main-box-5",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": " main-box-6",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Powered by ",
                  <a
                    href="http://www.coindesk.com/price"
>
                    CoinDesk
</a>,
                ],
                "className": "credits",
              },
              "ref": null,
              "rendered": Array [
                "Powered by ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "CoinDesk",
                    "href": "http://www.coindesk.com/price",
                  },
                  "ref": null,
                  "rendered": "CoinDesk",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "footer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
